Optionally remove dependency on golang.org/x/sys/unix without "appengine" build tag

I'm trying to use [gopherjs](https://github.com/gopherjs/gopherjs) to compile some Go code into JS. It works great for everything I've tried so far except logrus. The issue is that logrus depends on `golang.org/x/sys/unix` through `golang.org/x/crypto/ssh/terminal`. If we can use a compile tag to disable this dependency, that would be awesome.

I see that setting `--tags appengine` does fix the problem for logrus, but it interferes with building of other dependencies we have. (we are obviously not using appengine)

Example for testing:

`main.go:`
```go
package main

import (
	"github.com/gopherjs/gopherjs/js"
	"github.com/sirupsen/logrus"
)

func main() {
	js.Global.Set("run", func() {
		logrus.Info("example")
	})
	js.Global.Call("run")
}
```

`gopherjs build main.go`
```
../vendor/golang.org/x/sys/unix/syscall_unix.go:153:38: undeclared name: _Socklen
../vendor/golang.org/x/sys/unix/affinity_linux.go:13:20: undeclared name: _CPU_SETSIZE
../vendor/golang.org/x/sys/unix/affinity_linux.go:13:35: undeclared name: _NCPUBITS
../vendor/golang.org/x/sys/unix/affinity_linux.go:16:25: undeclared name: cpuMask
../vendor/golang.org/x/sys/unix/affinity_linux.go:49:27: undeclared name: cpuMask
../vendor/golang.org/x/sys/unix/flock.go:13:30: undeclared name: SYS_FCNTL
../vendor/golang.org/x/sys/unix/flock.go:16:42: undeclared name: Flock_t
../vendor/golang.org/x/sys/unix/sockcmsg_linux.go:14:29: undeclared name: Ucred
../vendor/golang.org/x/sys/unix/sockcmsg_unix.go:43:9: undeclared name: Cmsghdr
../vendor/golang.org/x/sys/unix/sockcmsg_linux.go:27:54: undeclared name: Ucred
../vendor/golang.org/x/sys/unix/sockcmsg_linux.go:27:54: too many errors
```

`gopherjs build main.go --tags appengine`
(success)

PR coming up. I'll add another build tag, "gopherjs" and it will work the same way as the appengine tag, but won't cause breakage for us by forcing us to compile other dependencies with the appengine tag.

--- a/terminal_bsd.go
+++ b/terminal_bsd.go
@@ -1,5 +1,5 @@
 // +build darwin freebsd openbsd netbsd dragonfly
-// +build !appengine
+// +build !appengine,!gopherjs
 
 package logrus
 

--- a/terminal_check_appengine.go
+++ b/terminal_check_appengine.go
@@ -1,4 +1,4 @@
-// +build appengine
+// +build appengine gopherjs
 
 package logrus
 

--- a/terminal_check_notappengine.go
+++ b/terminal_check_notappengine.go
@@ -1,4 +1,4 @@
-// +build !appengine
+// +build !appengine,!gopherjs
 
 package logrus
 

--- a/terminal_linux.go
+++ b/terminal_linux.go
@@ -3,7 +3,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build !appengine
+// +build !appengine,!gopherjs
 
 package logrus
 

